frontend/sw360-portlet/src/main/webapp/html/projects/includes/attachmentSelectTable.jspf

```diff
@@ -122,25 +152,206 @@
+                    initializeTableForLicenseInfo($linkedProjectsInfoTable);
+                } else if(data.tableMode == data.tableModeSourceBundle) {
+                    initializeTableForSourceBundle($linkedProjectsInfoTable);
+                } else {
+                    $linkedProjectsInfoTable.before($('<div/>', {
+                        'class': 'error foregroundAlert',
+                        text: 'Invalid table mode [' + data.tableMode + ']. Must be one of [' + data.tableModeLicenseInfo + ', ' + data.tableModeSourceBundle + '].'
+                    }));
+                    $linkedProjectsInfoTable.remove();
+                }
+
+                $("#selectAllCheckbox").on("change", function() {
+                    // first uncheck all in any case because on check only the default ones should be checked...
+                    $linkedProjectsInfoTable.find(":checkbox:not(#selectAllCheckbox)").prop('checked', false).trigger('change');
+                    // ...and check them if selected
+                    if (this.checked){
+                        $linkedProjectsInfoTable.find(":checkbox.defaultChecked").prop('checked', true).trigger('change');
+                    }
+                });
+
+                // helper functions
+                function initializeTableForSourceBundle($linkedProjectsInfoTable) {
+                    $linkedProjectsInfoTable.treetable({
+                        expandable: true,
+                        column: 1,
+                        initialState: "expanded"
+                    });
+                }
+
+                function initializeTableForLicenseInfo($linkedProjectsInfoTable) {
+                    // we can not use expanded as initial state because we have branches that are dynamically loaded...
+                    $linkedProjectsInfoTable.treetable({
+                        expandable: true,
+                        column: 1,
+                        initialState: "collapsed",
+                        onNodeExpand: function() { expandLicenseFile($linkedProjectsInfoTable, this); }
+                    });
+                    // ... therefore expand manually the loaded branches (projects and releases)
+                    $linkedProjectsInfoTable.find('tr.branch').each(function(index, element) {
+                        var data = $(element).data();
+
+                        if(data.rowType !== 'attachment') {
+                            $linkedProjectsInfoTable.treetable("expandNode", data.ttId);
+                        }
+                    });
+                }
+
+                function expandLicenseFile($linkedProjectsInfoTable, node) {
+                    var url,
+                        urlData = $linkedProjectsInfoTable.data(),
+                        data = node.row.data();
+
+                    if(data.rowType !== 'attachment' || data.loaded) {
+                        return;
+                    }
+                    data.loaded = true;
```

Comment:
Here, you set `data.loaded` to true before it is actually loaded. I'm not sure if this can create a problem.